layers/gpu_validation/gpu_subclasses.h

```diff
@@ -188,10 +133,83 @@ struct AccelerationStructureBuildValidationInfo {
+
+    
+    DeviceMemoryBlock output_mem_block;
+   
+    VkDescriptorSet output_buffer_desc_set = VK_NULL_HANDLE;  
+    VkDescriptorPool output_buffer_desc_pool = VK_NULL_HANDLE;
+    VkPipelineBindPoint pipeline_bind_point = VK_PIPELINE_BIND_POINT_MAX_ENUM;
+    bool uses_robustness = false;  // Only used in AnalyseAndeGenerateMessages, to output using LogWarning instead of LogError. It needs to be removed
+    vvl::Func command = vvl::Func::Empty;  // Should probably use Location instead
+    uint32_t desc_binding_index = vvl::kU32Max;// desc_binding is only used to help generate an error message
+    std::vector<DescBindingInfo> *desc_binding_list = nullptr;
+};
+
+class PreDrawResources : public CommandResources {
+  public:
+    ~PreDrawResources() {}
+
+    VkDescriptorPool desc_pool = VK_NULL_HANDLE;
+    VkDescriptorSet desc_set = VK_NULL_HANDLE;
+    VkBuffer indirect_buffer = VK_NULL_HANDLE;
+    VkDeviceSize indirect_buffer_offset = 0;
+    uint32_t indirect_buffer_stride = 0;
+    VkDeviceSize indirect_buffer_size = 0;
+    static constexpr uint32_t push_constant_words = 4;
+
+    void Destroy(gpuav::Validator &validator) final;
+    bool LogValidationMessage(gpuav::Validator &validator, VkQueue queue, VkCommandBuffer cmd_buffer, const uint32_t *debug_record,
+                              const uint32_t operation_index, const LogObjectList &objlist);
+};
+
+class PreDispatchResources : public CommandResources {
+  public:
+    ~PreDispatchResources() {}
+
+    VkDescriptorPool desc_pool = VK_NULL_HANDLE;
+    VkDescriptorSet desc_set = VK_NULL_HANDLE;
+    VkBuffer indirect_buffer = VK_NULL_HANDLE;
+    VkDeviceSize indirect_buffer_offset = 0;
+    static constexpr uint32_t push_constant_words = 4;
+
+    void Destroy(gpuav::Validator &validator) final;
+    bool LogValidationMessage(gpuav::Validator &validator, VkQueue queue, VkCommandBuffer cmd_buffer, const uint32_t *debug_record,
+                              const uint32_t operation_index, const LogObjectList &objlist);
+};
+
+class PreTraceRaysResources : public CommandResources {
+  public:
+    ~PreTraceRaysResources() {}
+
+    VkDescriptorPool desc_pool = VK_NULL_HANDLE;
+    VkDescriptorSet desc_set = VK_NULL_HANDLE;
+    VkDeviceAddress indirect_data_address = 0;
+    static constexpr uint32_t push_constant_words = 5;
+
+    void Destroy(gpuav::Validator &validator) final;
+    bool LogValidationMessage(gpuav::Validator &validator, VkQueue queue, VkCommandBuffer cmd_buffer, const uint32_t *debug_record,
+                              const uint32_t operation_index, const LogObjectList &objlist);
+};
+
 class CommandBuffer : public gpu_tracker::CommandBuffer {
   public:
-    // per draw/dispatch command state
-    std::vector<CommandInfo> per_draw_command_infos;
+    // per validated command state
+    std::vector<std::unique_ptr<CommandResources>> per_command_infos_2;
```

Comment:
```suggestion
    std::vector<std::unique_ptr<CommandResources>> per_command_infos;
```

the `2` is confusing, this isn't an API, we shouldn't need to 2 version of anything

Reply:
That is a leftover, forgot to rename it